src/OpenTelemetry.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -119,6 +119,10 @@ HRESULT STDMETHODCALLTYPE CorProfiler::Initialize(IUnknown* cor_profiler_info_un
     if (SUCCEEDED(hr))
     {
         Logger::Debug("Interface ICorProfilerInfo10 found.");
+        // .NET Core applications should use the dotnet startup hook to bootstrap OpenTelemetry so that the
+        // necessary dependencies will be available. Bootstrapping with the profiling APIs occurs to early
+        // and the necessary dependencies are not available yet.
+        use_dotnet_startuphook_bootstrapper = true;
```

Comment:
Checking for the v10 interface seems to be the safest way to determine if .NET Core is used.